File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:

       
        n = len(grid)
        sum_value = ((n**2)*((n**2)+1))//2
        # print(sum_value)
        values_set = set()
        duplicate_value = 0
        for i in range(0,n):
            for j in range(0,n):
                if grid[i][j] in values_set:
                    duplicate_value = grid[i][j]
                    continue
                sum_value-=grid[i][j]
                values_set.add(grid[i][j])
        return [duplicate_value, sum_value]

File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        
        int n = grid.length;
        double Cell_count = Math.pow(n,2);
        int sum_val = (int)(Cell_count*(Cell_count+1))/2;
        // System.out.println((int)Cell_count);
        HashSet<Integer> hashset = new HashSet<Integer>();
        
        int[] arr = new int[2];
        for(int i = 0; i<n;i++){
            for(int j =0;j<n;j++){
                if(hashset.contains(grid[i][j])){
                    arr[0] = grid[i][j];
                    continue;
                }
                sum_val-=grid[i][j];
                hashset.add(grid[i][j]);
            }
        }
        arr[1] = sum_val;
        return arr;
    }
}